src/main/java/de/unistuttgart/iste/gits/quiz_service/persistence/dao/ClozeElementEmbeddable.java

```diff
@@ -0,0 +1,28 @@
+package de.unistuttgart.iste.gits.quiz_service.persistence.dao;
+
+import de.unistuttgart.iste.gits.common.resource_markdown.ResourceMarkdownEmbeddable;
+import de.unistuttgart.iste.gits.common.resource_markdown.ResourceMarkdownEntity;
+import de.unistuttgart.iste.gits.generated.dto.ClozeElementType;
+import jakarta.persistence.*;
+import lombok.*;
+
+@Embeddable
+@Data
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+public class ClozeElementEmbeddable {
+
+    @Enumerated(EnumType.ORDINAL)
+    private ClozeElementType type;
+
+    @Embedded
+    @Builder.Default
+    private ResourceMarkdownEmbeddable text = new ResourceMarkdownEmbeddable("");
+
+    @Column(nullable = true)
+    private String correctAnswer;
```

Comment:
Wouldn't doing it like this mean that at the end of each text block there has to be a gap to fill in or am i misunderstanding how this works?

Reply:
youre misunderstanding it. you can check the schema, should make it clear

Reply:
so in the schema you have a union so there you differentiate between question elements and answer elements and in the db you have a single embeddable which can store both, am i understanding correctly?

Reply:
Yes, because adding inheritance in db is always a bit annoying and requires joins and stuff

Reply:
yes yes, it's fine i just wanted to make sure i understand everything correctly